Sources/NIO/EventLoopFuture.swift

```diff
@@ -183,6 +183,24 @@ public struct EventLoopPromise<Value> {
     public func fail(_ error: Error) {
         self._resolve(value: .failure(error))
     }
+    
+    /// Complete the promise with the passed in `EventLoopFuture<Value>`.
+    ///
+    /// This method is equivalent to invoking `future.cascade(to: promise)`,
+    /// but sometimes may read better than its cascade counterpart.
+    /// 
+    /// - parameters:
+    ///     - future: The future whose value will be used to succeed or fail this promise.
+    /// - seealso: `EventLoopFuture.cascade(to:)`
+    @inlinable
+    public func completeWith(_ future: EventLoopFuture<Value>) {
+        future.whenComplete { result in
+            switch result {
+            case let .success(value): self.succeed(value)
+            case let .failure(error): self.fail(error)
+            }
+        }
+    }
```

Comment:
no strong opinion about expressing it in line OR delegating to `future.cascade(to: self)`, let me know if better to delegate.

Reply:
Heh, I just wrote that comment. Let's delegate.

Reply:
ok :)